src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -538,80 +541,11 @@ public static ulong GetKey(ReadOnlySpan<byte> propertyName)
             return key;
         }
 
-        // Return the element type of the IEnumerable or return null if not an IEnumerable.
-        public static Type GetElementType(Type propertyType)
-        {
-            if (!typeof(IEnumerable).IsAssignableFrom(propertyType))
-            {
-                return null;
-            }
-
-            // Check for Array.
-            Type elementType = propertyType.GetElementType();
-            if (elementType != null)
-            {
-                return elementType;
-            }
-
-            // Check for Dictionary<TKey, TValue> or IEnumerable<T>
-            if (propertyType.IsGenericType)
-            {
-                Type[] args = propertyType.GetGenericArguments();
-
-                if (IsDictionaryClassType(propertyType) &&
-                    args.Length >= 2 && // It is >= 2 in case there is a IDictionary<TKey, TValue, TSomeExtension>.
-                    args[0].UnderlyingSystemType == typeof(string))
-                {
-                    return args[1];
-                }
-
-                if (typeof(IEnumerable).IsAssignableFrom(propertyType) && args.Length >= 1) // It is >= 1 in case there is an IEnumerable<T, TSomeExtension>.
-                {
-                    return args[0];
-                }
-            }
-
-            return typeof(object);
-        }
-
-        public static ClassType GetClassType(Type type, bool checkForConverter, JsonSerializerOptions options)
-        {
-            Debug.Assert(type != null);
-
-            if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Nullable<>))
-            {
-                type = Nullable.GetUnderlyingType(type);
-                checkForConverter = true;
-            }
-
-            if (type == typeof(object))
-            {
-                return ClassType.Unknown;
-            }
-
-            if (checkForConverter && options.HasConverter(type))
-            {
-                return ClassType.Value;
-            }
-
-            if (type.IsArray)
-            {
-                return ClassType.Enumerable;
-            }
-
-            if (IsDictionaryClassType(type))
-            {
-                return ClassType.Dictionary;
-            }
-
-            if (typeof(IEnumerable).IsAssignableFrom(type))
-            {
-                return ClassType.Enumerable;
-            }
-
-            return ClassType.Object;
-        }
-
+        // This method gets the runtime information for a given type or property.
+        // The runtime information consists of the class type, runtime type, element type (if the type is a collection),
+        // the underlying type (if the type is nullable type e.g. int?), the add method (if the type i
```

Comment:
nit: "is" instead of "i" for "if the type i".

Also perhaps use a new line for each item (type, runtime type, elementtype, underlying type, add,)